presto-base-jdbc/src/test/java/io/prestosql/plugin/jdbc/TestCachingJdbcClient.java

```diff
@@ -14,15 +14,176 @@
 package io.prestosql.plugin.jdbc;
 
 import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import io.airlift.units.Duration;
+import io.prestosql.spi.connector.ColumnMetadata;
+import io.prestosql.spi.connector.ConnectorSession;
+import io.prestosql.spi.connector.ConnectorTableMetadata;
+import io.prestosql.spi.connector.SchemaTableName;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
 import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
 import java.util.Set;
 
+import static com.google.common.collect.Sets.difference;
 import static io.prestosql.spi.testing.InterfaceTestUtils.assertAllMethodsOverridden;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static java.util.Collections.emptyList;
+import static java.util.concurrent.TimeUnit.DAYS;
+import static org.assertj.core.api.Assertions.assertThat;
 
+@Test(singleThreaded = true)
 public class TestCachingJdbcClient
 {
+    private static final Duration FOREVER = Duration.succinctDuration(1, DAYS);
+    private static final ConnectorSession session = testSessionBuilder().build().toConnectorSession();
+    private static final JdbcIdentity identity = JdbcIdentity.from(session);
+
+    private TestingDatabase database;
+    private CachingJdbcClient cachingJdbcClient;
+    private JdbcClient dbHacker;
+    private String schema;
+
+    @BeforeMethod
+    public void setUp()
+            throws Exception
+    {
+        database = new TestingDatabase();
+        cachingJdbcClient = createCachingJdbcClient(true);
+        dbHacker = database.getJdbcClient();
+        schema = dbHacker.getSchemaNames(identity).iterator().next();
+    }
+
+    private CachingJdbcClient createCachingJdbcClient(boolean cacheMissing)
+    {
+        return new CachingJdbcClient(database.getJdbcClient(), FOREVER, cacheMissing);
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown()
+            throws Exception
+    {
+        database.close();
+    }
+
+    @Test
+    public void testSchemaNamesCached()
+    {
+        String phantomSchema = "phantom_schema";
+
+        dbHacker.createSchema(identity, phantomSchema);
+        Set<String> cachedSchemas = cachingJdbcClient.getSchemaNames(identity);
+        dbHacker.dropSchema(identity, phantomSchema);
+
+        Sets.SetView<String> actualDbState = difference(cachedSchemas, Set.of(phantomSchema));
```

Comment:
Declare type as View. Instead of using difference please use `jdbcClient.getSchemaNames()` and just verify that cached schemas do not contain `phantom_schema`

Reply:
This was a way to express test via `actualDbState`. With `contains` and `doesNotContain` there is no sense in this variable at all. However it seems that despite it's less verbose it's still expressive.